### PR TITLE
Merge alpine_3.9* dockerfiles

### PR DESCRIPTION
**Pull request by avtikhon**
Created 9 commits to make dockerfiles/alpine_3.9_*.x files looks the same
and merged it into single common file dockerfiles/alpine_3.9.

### CODE REVIEW

**Comment by LeonidVas** [comment]
Hi! Thank you for the patch.
I have some comments / questions:
* Please, split it into two commits (merge dockerfiles and add LUAROCKS_INSTALL var.)
* I find it strage that we want to use a luarocks instead of tarantoolctl rocks. What is it for? Who is the customer?

**Comment by avtikhon** [comment]
Thanks, for the review, please check my comments below:

> Hi! Thank you for the patch.
> I have some comments / questions:
> 
> * Please, split it into two commits (merge dockerfiles and add LUAROCKS_INSTALL var.)

Not really possible, due to:
- merged dockerfiles would have 'tarantoolctl rocks' either 'luarocks' only w/o LUAROCKS_INSTALL variable and some of the builds would fail;
- sure the implementation of the LUAROCKS_INSTALL variable as the first commit would fix the issue mentioned above, but in this way the feature will be implemented w/o it's use - dead code.

> * I find it strage that we want to use a luarocks instead of tarantoolctl rocks. What is it for? Who is the customer?

In real we need to use 'tarantooctl rocks' command to install rocks, but not all of the Tarantool versions has it workable - in this way we need to use 'luarocks' as workaround.

**Comment by LeonidVas** [comment]
> Not really possible, due to:
> 
> * merged dockerfiles would have 'tarantoolctl rocks' either 'luarocks' only w/o LUAROCKS_INSTALL variable and some of the builds would fail;
> * sure the implementation of the LUAROCKS_INSTALL variable as the first commit would fix the issue mentioned above, but in this way the feature will be implemented w/o it's use - dead code. If this solution is ok, I'll do it.

Already discussed, just for a note: "1st commit - add var, 2nd - merge".

**Comment by LeonidVas** [comment]
> In real we need to use 'tarantooctl rocks' command to install rocks, but not all of the Tarantool versions has it workable - in this way we need to use 'luarocks' as workaround.

Ok. I am surprised.

**Comment by avtikhon** [comment]
> > Not really possible, due to:
> > 
> > * merged dockerfiles would have 'tarantoolctl rocks' either 'luarocks' only w/o LUAROCKS_INSTALL variable and some of the builds would fail;
> > * sure the implementation of the LUAROCKS_INSTALL variable as the first commit would fix the issue mentioned above, but in this way the feature will be implemented w/o it's use - dead code. If this solution is ok, I'll do it.
> 
> Already discussed, just for a note: "1st commit - add var, 2nd - merge".

Right, but looking deeply found that all the changes were better to split.

**Comment by avtikhon** [comment]
> > In real we need to use 'tarantooctl rocks' command to install rocks, but not all of the Tarantool versions has it workable - in this way we need to use 'luarocks' as workaround.
> 
> Ok. I am surprised.

Unfortunately yes.

File: .gitlab.mk [hunk 1]
```diff
@@ -1,9 +1,14 @@
 TNT_VER=$(shell cat versions/${OS}_${DIST}_${VER})
+ROCKS_INSTALLER?='tarantoolctl rocks'
 
 build:
-	docker build --no-cache --network=host --build-arg TNT_VER=${TNT_VER} \
+	[ "${DOCKERFILE_NAME_SUFFIX}" != "" ] && \
+		DOCKERFILE_SUFFIX=_${DOCKERFILE_NAME_SUFFIX} ; \
```
Comment (LeonidVas) [inline .gitlab.mk:9]: It looks like the changes in the work with the DOCKERFILE_SUFFIX are not related to the commit theme and can be moved to a separate commit.

File: .gitlab.mk [hunk 2]
```diff
@@ -1,9 +1,14 @@
 TNT_VER=$(shell cat versions/${OS}_${DIST}_${VER})
```
Comment (LeonidVas) [inline .gitlab.mk:1]: May be add information to the commit message that the commit is a preparation for the merge of dockerfiles? Up to you.

File: dockerfiles/alpine_3.5_2.x [hunk 3]
```diff
@@ -195,7 +195,8 @@ RUN set -x \
     && : "---------- luarocks ----------" \
     && cd / \
     && : "ldoc" \
-    && ${ROCKS_INSTALLER} install ldoc --server=http://rocks.moonscript.org \
+    && ( ${ROCKS_INSTALLER} install ldoc \
```
Comment (LeonidVas) [inline dockerfiles/alpine_3.5_2.x:198]: You are changing behavior. Before: we will try to install "ldoc" only from the http://rocks.moonscript.org server. Now: we will try to install "ldoc" from the "default" server (from the config file) and only in case of failure from http://rocks.moonscript.org. is it ok?

**Comment by LeonidVas** [comment]
> > > Not really possible, due to:
> > > 
> > > * merged dockerfiles would have 'tarantoolctl rocks' either 'luarocks' only w/o LUAROCKS_INSTALL variable and some of the builds would fail;
> > > * sure the implementation of the LUAROCKS_INSTALL variable as the first commit would fix the issue mentioned above, but in this way the feature will be implemented w/o it's use - dead code. If this solution is ok, I'll do it.
> > 
> > 
> > Already discussed, just for a note: "1st commit - add var, 2nd - merge".
> 
> Right, but looking deeply found that all the changes were better to split.

Great job!
See the comments per commit below:
* "Set rocks tool installer using variable" - may be add information to the commit message that the commit is a preparation for the merge of dockerfiles? Up to you.
* "Use LUAROCKS_INSTALL for dockerfiles builds" - (Part of ?) LGTM.
* "Use LUAROCKS_INSTALL for rest of dockerfiles" - LGTM.
* "Add server path for luarocks ldoc installation" - (Part of ?). See a question in the code.
* "Enable ENABLE_BUNDLED_LIBYAML flag" - LGTM.
* "Enable manual build with makefile" - local host -> localhost? How can I run this? Can I do it locally?
* "Remove perl installation from build depends" - Why Perl has become obsolete? Are you disappointed in Perl?:-)
* "Add unzip tool to rocks installations" - LGTM
* "Use libressl in Alpine based images builds" - the policy of the tarantool team - to use an openssl library instead of libressl. Have you decided to move against the current? You must have a good reasoned position. Please, list all your arguments.
"Merge same Alpine 3.9 dockerfiles" - LGTM

File: .gitlab.mk (hunk 1 above)
Comment (avtikhon) [reply .gitlab.mk:9]: > It looks like the changes in the work with the DOCKERFILE_SUFFIX are not related to the commit theme and can be moved to a separate commit.

Right, moved to "Enable manual build with makefile"  commit.

**Comment by LeonidVas** [comment]
> * "Set rocks tool installer using variable" - may be add information to the commit message that the commit is a preparation for the merge of dockerfiles? Up to you

LGTM, after split commit.

**Comment by avtikhon** [comment]
> > > > Not really possible, due to:
> > > > 
> > > > * merged dockerfiles would have 'tarantoolctl rocks' either 'luarocks' only w/o LUAROCKS_INSTALL variable and some of the builds would fail;
> > > > * sure the implementation of the LUAROCKS_INSTALL variable as the first commit would fix the issue mentioned above, but in this way the feature will be implemented w/o it's use - dead code. If this solution is ok, I'll do it.
> > > 
> > > 
> > > Already discussed, just for a note: "1st commit - add var, 2nd - merge".
> > 
> > 
> > Right, but looking deeply found that all the changes were better to split.
> 
> Great job!
> See the comments per commit below:
> 
> * "Set rocks tool installer using variable" - may be add information to the commit message that the commit is a preparation for the merge of dockerfiles? Up to you.

Seems not really needed, due to merge is not the cause of the change, decided to add more info.

> * "Use LUAROCKS_INSTALL for dockerfiles builds" - (Part of ?) LGTM.
> * "Use LUAROCKS_INSTALL for rest of dockerfiles" - LGTM.
> * "Add server path for luarocks ldoc installation" - (Part of ?). See a question in the code.

Work in progress - will comment here later.

> * "Enable ENABLE_BUNDLED_LIBYAML flag" - LGTM.
> * "Enable manual build with makefile" - local host -> localhost? How can I run this? Can I do it locally?

Right, in previous to rerun the build too many variables had to be manually took from gitlab-ci yaml configuration.
After this patch make file can be run manually just with minimal number of variables.

> * "Remove perl installation from build depends" - Why Perl has become obsolete? Are you disappointed in Perl?:-)

Perl removed from build time installation part, it means that it will be removed just after the build anyway.

> * "Add unzip tool to rocks installations" - LGTM
> * "Use libressl in Alpine based images builds" - the policy of the tarantool team - to use an openssl library instead of libressl. Have you decided to move against the current? You must have a good reasoned position. Please, list all your arguments.
>   "Merge same Alpine 3.9 dockerfiles" - LGTM

File: dockerfiles/alpine_3.5_2.x (hunk 3 above)
Comment (avtikhon) [reply dockerfiles/alpine_3.5_2.x]: Decided to remove the patch from patch set.

Comment (LeonidVas) [reply dockerfiles/alpine_3.5_2.x:198]: Patch is back)

**Comment by LeonidVas** [comment]
LGTM. As I understand it, we need to push # 177 to get the green pipeline in gitlab. Please, make sure the build works correctly in gitlab before pushing this patchset.
